okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpChannelBuilder.java

```diff
@@ -144,6 +158,19 @@ public OkHttpChannelBuilder connectionSpec(ConnectionSpec connectionSpec) {
   protected ChannelEssentials buildEssentials() {
     final ExecutorService executor = (transportExecutor == null)
         ? SharedResourceHolder.get(DEFAULT_TRANSPORT_THREAD_POOL) : transportExecutor;
+    switch (negotiationType) {
+      case TLS:
+        if (sslSocketFactory == null) {
+          sslSocketFactory = (SSLSocketFactory) SSLSocketFactory.getDefault();
+        }
+        break;
+      case PLAINTEXT:
+        sslSocketFactory = null;
```

Comment:
This change all happens on channel builder, the transport's behavior is still "using TLS if we get a SSLSocketFactory".

So if user calls negotiationType(NegotiationType.PLAINTEXT), I clear the sslSocketFactory to honor this call even if the user called sslSocketFactory() previously.

Actually, I call negotiationType(NegotiationType.TLS) inside sslSocketFactory() for the same reason: honor the call even if the user called negotiationType(NegotiationType.PLAINTEXT) previously.